### PR TITLE
Support `createDatabaseIfNotExist`

### PR DESCRIPTION
**Pull request by mirromutth**
Motivation:

See also #158 .

It should be named `createDatabaseIfNotExist`, the same as JDBC naming.

Modification:

- [x] Add option to enable `createDatabaseIfNotExist`
- [x] Support `COM_INIT_DB` and database can be initialized after handshake
- [x] Support sending `CREATE DATABASE IF NOT EXISTS` statement and retrying `COM_INIT_DB` after `COM_INIT_DB` fails for the first time

Result:

Support `createDatabaseIfNotExist`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,19 +66,19 @@
...
-    <jmh.version>1.36</jmh.version>
-    <junit.version>5.9.2</junit.version>
-    <logback.version>1.3.6</logback.version>
-    <mockito.version>4.0.0</mockito.version>
-    <mysql.version>8.0.32</mysql.version>
-    <testcontainers.version>1.17.6</testcontainers.version>
-    <hikari-cp.version>4.0.3</hikari-cp.version>
-    <spring-framework.version>5.3.25</spring-framework.version>
-    <jackson.version>2.14.2</jackson.version>
+    <assertj.version>3.24.2</assertj.version>
```
Comment (JohnNiang) [inline pom.xml]: How about upgrading the versions of these dependencies in antoher PR? IMO, those changes are not directly related to the current PR. I prefer to do only one thing in a PR. Of course, this will not prevent the current PR from being merged.

Comment (mirromutth) [reply pom.xml]: Yes, you are right, I will create a new PR for upgrading dependencies.

File: src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 2]
```diff
@@ -245,29 +256,31 @@ public boolean equals(Object o) {
...
-                   ", tcpNoDelay=" + tcpNoDelay + ", tcpKeepAlive=" + tcpKeepAlive + ", connectTimeout=" +
-                   connectTimeout + ", socketTimeout=" + socketTimeout + ", serverZoneId=" + serverZoneId +
-                   ", zeroDateOption=" + zeroDateOption + ", user='" + user + '\'' + ", password=" + password +
-                   ", database='" + database + "', preferPrepareStatement=" + preferPrepareStatement +
-                   ", queryCacheSize=" + queryCacheSize + ", prepareCacheSize=" + prepareCacheSize +
-                   ", extensions=" + extensions + ", passwordPublisher=" + passwordPublisher + '}';
+                ", tcpNoDelay=" + tcpNoDelay + ", tcpKeepAlive=" + tcpKeepAlive + ", connectTimeout=" +
+                connectTimeout + ", socketTimeout=" + socketTimeout + ", serverZoneId=" + serverZoneId +
+                ", zeroDateOption=" + zeroDateOption + ", user='" + user + '\'' + ", password=" + password +
+                ", database='" + database + ", createDatabaseIfNotExist=" + createDatabaseIfNotExist +
```
Comment (jchrys) [inline src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]: seems closing `'` is misplaced.
```suggestion
                ", database='" + database + "', createDatabaseIfNotExist=" + createDatabaseIfNotExist +
```

File: src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 3]
```diff
@@ -245,29 +256,31 @@ public boolean equals(Object o) {
...
+                "', preferPrepareStatement=" + preferPrepareStatement + ", queryCacheSize=" + queryCacheSize +
```
Comment (jchrys) [inline src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]: seems closing ' is misplaced.
```suggestion
                ", preferPrepareStatement=" + preferPrepareStatement + ", queryCacheSize=" + queryCacheSize +
```

File: src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 4]
```diff
@@ -245,29 +256,31 @@ public boolean equals(Object o) {
...
 
         return "MySqlConnectionConfiguration{, unixSocket='" + domain + "', connectTimeout=" +
-               connectTimeout + ", socketTimeout=" + socketTimeout + ", serverZoneId=" + serverZoneId +
-               ", zeroDateOption=" + zeroDateOption + ", user='" + user + "', password=" + password +
-               ", database='" + database + "', preferPrepareStatement=" + preferPrepareStatement +
-               ", queryCacheSize=" + queryCacheSize + ", prepareCacheSize=" + prepareCacheSize +
-               ", extensions=" + extensions + ", passwordPublisher=" + passwordPublisher + '}';
+            connectTimeout + ", socketTimeout=" + socketTimeout + ", serverZoneId=" + serverZoneId +
+            ", zeroDateOption=" + zeroDateOption + ", user='" + user + "', password=" + password +
+            ", database='" + database + ", createDatabaseIfNotExist=" + createDatabaseIfNotExist +
```
Comment (jchrys) [inline src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]: seems closing ' is misplaced.
```suggestion
            ", database='" + database + "', createDatabaseIfNotExist=" + createDatabaseIfNotExist +
```

File: src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 5]
```diff
@@ -245,29 +256,31 @@ public boolean equals(Object o) {
...
+            "', preferPrepareStatement=" + preferPrepareStatement + ", queryCacheSize=" + queryCacheSize +
```
Comment (jchrys) [inline src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]: seems closing ' is misplaced.
```suggestion
            ", preferPrepareStatement=" + preferPrepareStatement + ", queryCacheSize=" + queryCacheSize +
```

**Review by jchrys — COMMENTED** [review]
Would you mind adding a brief description of the newly added feature to the README file?

below are just minor details

Thanks a lot!

**Review by jchrys — APPROVED** [review]
lgtm
